server/src/main/java/org/apache/druid/server/coordinator/duty/HistoricalCloningDuty.java

```diff
@@ -0,0 +1,93 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.coordinator.duty;
+
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.server.coordinator.DruidCoordinatorRuntimeParams;
+import org.apache.druid.server.coordinator.ServerHolder;
+import org.apache.druid.server.coordinator.loading.SegmentAction;
+import org.apache.druid.server.coordinator.stats.CoordinatorRunStats;
+import org.apache.druid.server.coordinator.stats.Dimension;
+import org.apache.druid.server.coordinator.stats.RowKey;
+import org.apache.druid.server.coordinator.stats.Stats;
+import org.apache.druid.timeline.DataSegment;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HistoricalCloningDuty implements CoordinatorDuty
+{
+  private static final Logger log = new Logger(HistoricalCloningDuty.class);
+
+  @Nullable
+  @Override
+  public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
+  {
+    final Map<String, String> cloneServers = params.getCoordinatorDynamicConfig().getCloneServers();
+    if (cloneServers.isEmpty()) {
+      return params;
+    }
+
+    final CoordinatorRunStats stats = params.getCoordinatorStats();
+    // TODO: clean up
+    final Map<String, ServerHolder> historicalMap = params.getDruidCluster()
+                                                          .getHistoricals()
+                                                          .values()
+                                                          .stream()
+                                                          .flatMap(Collection::stream)
+                                                          .collect(Collectors.toMap(
+                                                              serverHolder -> serverHolder.getServer().getHost(),
+                                                              serverHolder -> serverHolder
+                                                          ));
+
+    for (Map.Entry<String, String> entry : cloneServers.entrySet()) {
+      String sourceHistoricalName = entry.getKey();
+      ServerHolder sourceServer = historicalMap.get(sourceHistoricalName);
+
+      String targetHistorical = entry.getValue();
+      ServerHolder targetServer = historicalMap.get(targetHistorical);
+
+      for (DataSegment segment : sourceServer.getProjectedSegments().getSegments()) {
+        if (!targetServer.getServedSegments().contains(segment)) {
+          targetServer.getPeon().loadSegment(segment, SegmentAction.LOAD, null);
```

Comment:
Load and drop should go via the `LoadQueueManager`.
See `StrategicSegmentAssigner.loadSegment` for reference.